x-pack/plugins/search_playground/public/hooks/use_source_indices_field.ts

```diff
@@ -55,52 +51,35 @@ export const useSourceIndicesFields = () => {
   });
 
   const {
-    field: { onChange: onSourceFieldsChange },
-  } = useController({
-    name: ChatFormFields.sourceFields,
+    field: { onChange: onQueryFieldsOnChange, value: queryFields },
+  } = useController<ChatForm, ChatFormFields.queryFields>({
+    name: ChatFormFields.queryFields,
   });
 
-  const { data: fields } = useQuery({
-    enabled: selectedIndices.length > 0,
-    queryKey: ['fields', selectedIndices.toString()],
-    queryFn: async () => {
-      const response = await services.http.post<IndicesQuerySourceFields>(
-        APIRoutes.POST_QUERY_SOURCE_FIELDS,
-        {
-          body: JSON.stringify({ indices: selectedIndices }),
-        }
-      );
-      return response;
-    },
+  const {
+    field: { onChange: onSourceFieldsChange, value: sourceFields },
+  } = useController({
+    name: ChatFormFields.sourceFields,
   });
+  const { fields, isLoading: isFieldsLoading } = useIndicesFields(selectedIndices);
 
   useEffect(() => {
-    if (fields) {
-      resetField(ChatFormFields.queryFields);
-
+    if (fields && !isFieldsLoading) {
       const defaultFields = getDefaultQueryFields(fields);
       const defaultSourceFields = getDefaultSourceFields(fields);
 
-      const indicesWithNoSourceFields = getIndicesWithNoSourceFields(defaultSourceFields);
-
-      if (indicesWithNoSourceFields) {
-        setNoFieldsIndicesWarning(indicesWithNoSourceFields);
-      } else {
-        setNoFieldsIndicesWarning(null);
-      }
-
       onElasticsearchQueryChange(createQuery(defaultFields, defaultSourceFields, fields));
-      onSourceFieldsChange(defaultSourceFields);
+      onQueryFieldsOnChange(merge(defaultFields, queryFields));
+
+      onSourceFieldsChange(merge(defaultSourceFields, sourceFields));
       usageTracker?.count(
         AnalyticsEvents.sourceFieldsLoaded,
         Object.values(fields)?.flat()?.length
       );
-    } else {
-      setNoFieldsIndicesWarning(null);
```

Comment:
this warning is still important to highlight. Does this move somewhere else?

Reply:
Add new callout when index selected in flyout